### DISCUSSION TITLE
-preview resources requested

### DISCUSSION THREAD

**Discussion by chasemc** [nextflow-io/nextflow#3298] [Ideas]
This might be naive but would it be possible to render a small report on the resources that will be requested per process when using the new `-preview` flag? While it won't account for the number of spawned processes it would give a better idea of resources that will be requested.
Large workflows, especially written with @nf-core standards have mem/cpu/time definitions scattered across multiple files and end-users might not have a great idea about what resources are needed/will be requested?

**Comment by mahesh-panchal** [comment]
I think this information is embedded in the output of `nextflow config ...` but one would likely need to use a complex `jq` query to get the information out nicely.

I think nf-core has a better way of providing this information though (imo). All workflows run on a dataset that would be typically representative of an expected input, and this information is then present on each nf-core's workflow page. e.g.:
https://nf-co.re/rnaseq/results#rnaseq/results-e049f51f0214b2aef7624b9dd496a404a7c34d14/
This could certainly do with a better label than aws results, but you can then see a more realistic idea of the resources used including run-time etc, and the reports are also included in each folder. The initial configuration settings above may end up being quite different since often modules are just assigned a label, and very rarely adjusted afterwards.
